watertap/unit_models/MD/membrane_distillation_base.py

```diff
@@ -0,0 +1,741 @@
+                    t, b.cold_ch.length_domain.last()
+                ].flow_mass_phase_comp["Liq", "H2O"]
+            )
+
+        self._add_mass_transfer()
+        self._add_heat_transfer()
+
+        @self.Expression(
+            self.flowsheet().config.time,
+            doc="Average thermal efficiency",
+        )
+        def thermal_efficiency(b, t):
+            total_enth_flux = sum(
+                b.flux_enth_hot[t, x] for x in self.difference_elements
+            )
+            total_cond_heat_flux = sum(
+                b.flux_conduction_heat[t, x] for x in self.difference_elements
+            )
+
+            return total_enth_flux / (total_enth_flux + total_cond_heat_flux)
+
+        @self.Expression(
+            self.flowsheet().config.time,
+            doc="module heat recovery: ratio of the actual heat recovered by cold side to the maximum ideal heat recovery",
+        )
+        def effectiveness(b, t):
+
+            return (
+                b.cold_ch.properties[t, b.first_element].temperature
+                - b.cold_ch.properties[t, b.cold_ch.length_domain.last()].temperature
+            ) / (
+                b.hot_ch.properties[t, b.first_element].temperature
+                - b.cold_ch.properties[t, b.cold_ch.length_domain.last()].temperature
+            )
+
+    def _add_mass_flux(self):
+
+        solute_set = self.config.hot_ch.property_package.solute_set
+
+        units_meta = (
+            self.config.hot_ch.property_package.get_metadata().get_derived_units
+        )
+
+        # todo: add calculation method for permeability coefficient
+        self.permeability_coef = Var(
+            self.flowsheet().config.time,
+            initialize=1e-10,
+            bounds=(1e-11, 1e-9),
+            units=units_meta("mass")
+            * units_meta("time") ** -1
+            * units_meta("length") ** -1
+            * units_meta("pressure") ** -1,
+            doc="membrane permeabiity coeffcient",
+        )
+
+        self.flux_mass = Var(
+            self.flowsheet().config.time,
+            self.difference_elements,
+            initialize=1e-10,
+            bounds=(1e-12, 1e-1),
+            units=units_meta("mass")
+            * units_meta("time") ** -1
+            * units_meta("length") ** -2,
+            doc="solvent vapor mass flux across the membrane",
+        )
+
+        self.flux_enth_hot = Var(
+            self.flowsheet().config.time,
+            self.difference_elements,
+            initialize=1e3,
+            bounds=(1e-10, 1e5),
+            units=pyunits.J * pyunits.s**-1 * pyunits.m**-2,
+            doc="hot side evaporation enthalpy flux",
+        )
+
+        self.flux_enth_cold = Var(
+            self.flowsheet().config.time,
+            self.difference_elements,
+            initialize=1e3,
+            bounds=(1e-10, 1e5),
+            units=pyunits.J * pyunits.s**-1 * pyunits.m**-2,
+            doc="cold side condensation enthalpy flux",
+        )
```

Comment:
The variables directly above this use units_meta. Is there a reason that these variables do not follow the same convention? I believe this only occurs a few times across the document.